app/src/main/java/de/tum/in/tumcampusapp/services/SilenceService.java

```diff
@@ -96,47 +98,44 @@ protected void onHandleWork(@NonNull Intent intent) {
         if (am == null) {
             return;
         }
-        try (Cursor cursor = calendarManager.getCurrentFromDb()) {
-            Utils.log("Current lectures: " + cursor.getCount());
-
-            if (cursor.getCount() == 0 || isDoNotDisturbMode()) {
-                if (Utils.getInternalSettingBool(this, Const.SILENCE_ON, false) && !isDoNotDisturbMode()) {
-                    // default: old state
-                    Utils.log("set ringer mode to old state");
-                    am.setRingerMode(Integer.parseInt(
-                            Utils.getSetting(this, Const.SILENCE_OLD_STATE,
-                                             Integer.toString(AudioManager.RINGER_MODE_NORMAL))));
-                    Utils.setInternalSetting(this, Const.SILENCE_ON, false);
-
-                    Cursor cursor2 = calendarManager.getNextCalendarItem();
-                    if (cursor.getCount() != 0) { //Check if we have a "next" item in the database and update the refresh interval until then. Otherwise use default interval.
-                        // refresh when next event has started
-                        waitDuration = getWaitDuration(cursor2.getString(1));
-                    }
-                    cursor2.close();
-                }
-            } else {
-                // remember old state if just activated ; in doubt dont change
-                if (!Utils.getInternalSettingBool(this, Const.SILENCE_ON, true)) {
-                    Utils.setSetting(this, Const.SILENCE_OLD_STATE, am.getRingerMode());
+        List<CalendarItem> currentLectures = calendarManager.getCurrentFromDb();
+        Utils.log("Current lectures: " + currentLectures.size());
+
+        if (currentLectures.size() == 0 || isDoNotDisturbMode()) {
+            if (Utils.getInternalSettingBool(this, Const.SILENCE_ON, false) && !isDoNotDisturbMode()) {
+                // default: old state
+                Utils.log("set ringer mode to old state");
+                am.setRingerMode(Integer.parseInt(
+                        Utils.getSetting(this, Const.SILENCE_OLD_STATE,
+                                         Integer.toString(AudioManager.RINGER_MODE_NORMAL))));
+                Utils.setInternalSetting(this, Const.SILENCE_ON, false);
+
+                List<CalendarItem> nextCalendarItem = calendarManager.getNextCalendarItem();
+                if (nextCalendarItem.size() != 0) { //Check if we have a "next" item in the database and update the refresh interval until then. Otherwise use default interval.
+                    // refresh when next event has started
+                    waitDuration = getWaitDuration(nextCalendarItem.get(0).getDtstart());
```

Comment:
for this logic we could probably use a `getNextCalendarItem()` method that actually only returns an Optional of the very next item, instead of a List...

Reply:
Yeah, the naming is misleading, the query originally returned 4 items and the same query is used at `onRequestCard` function which uses the list.